src/tabmat/categorical_matrix.py

```diff
@@ -302,38 +351,56 @@ def __init__(
             self._has_missings = False
 
         self.drop_first = drop_first
-        self.shape = (len(self.cat), len(self.cat.categories) - int(drop_first))
-        self.indices = self.cat.codes.astype(np.int32)
-        self.x_csc: Optional[tuple[Optional[np.ndarray], np.ndarray, np.ndarray]] = None
+        self.indices = indices.astype(np.int32, copy=False)
+        self.shape = (len(self.indices), len(self.categories) - int(drop_first))
+        self.x_csc = None
         self.dtype = np.dtype(dtype)
 
         self._colname = column_name
+        self._colname_format = column_name_format
+
         if term_name is None:
             self._term = self._colname
         else:
             self._term = term_name
-        self._colname_format = column_name_format
 
     __array_ufunc__ = None
 
+    @property
+    def cat(self):
+        """Return a series with same data as what was initially fed to __init__.
+
+        This property is available for backward compatibility.
+        """
+        warnings.warn(
+            "This property will be removed in the next major release.",
+            category=DeprecationWarning,
+        )
+
+        if _is_polars(self._input_dtype):
```

Comment:
Am I mistaken that `_input_dtype` may neither be "polars" nor "pandas" if the initial array given to `__int__` was a plain NumPy array?

Reply:
That's in line with current behaviour and so backwards compatible. If the input is a NumPy array, we return a `pandas.Categorical`.